src/lib/logstore/log_dev.cpp

```diff
@@ -35,21 +35,18 @@ namespace homestore {
 
 SISL_LOGGING_DECL(logstore)
 
-#define THIS_LOGDEV_LOG(level, msg, ...) HS_SUBMOD_LOG(level, logstore, , "logdev", m_family_id, msg, __VA_ARGS__)
+#define THIS_LOGDEV_LOG(level, msg, ...) HS_SUBMOD_LOG(level, logstore, , "logdev", m_logdev_id, msg, __VA_ARGS__)
 #define THIS_LOGDEV_PERIODIC_LOG(level, msg, ...)                                                                      \
-    HS_PERIODIC_DETAILED_LOG(level, logstore, "logdev", m_family_id, , , msg, __VA_ARGS__)
+    HS_PERIODIC_DETAILED_LOG(level, logstore, "logdev", m_logdev_id, , , msg, __VA_ARGS__)
 
 static bool has_data_service() { return HomeStore::instance()->has_data_service(); }
 // static BlkDataService& data_service() { return HomeStore::instance()->data_service(); }
 
-LogDev::LogDev(const logstore_family_id_t f_id, const std::string& logdev_name) :
-        m_family_id{f_id}, m_logdev_meta{logdev_name} {
-    m_flush_size_multiple = 0;
-    if (f_id == LogStoreService::DATA_LOG_FAMILY_IDX) {
-        m_flush_size_multiple = HS_DYNAMIC_CONFIG(logstore->flush_size_multiple_data_logdev);
-    } else if (f_id == LogStoreService::CTRL_LOG_FAMILY_IDX) {
-        m_flush_size_multiple = HS_DYNAMIC_CONFIG(logstore->flush_size_multiple_ctrl_logdev);
-    }
+LogDev::LogDev(const logdev_id_t id) : m_logdev_id{id} {
+    m_flush_size_multiple = HS_DYNAMIC_CONFIG(logstore->flush_size_multiple_logdev);
+    register_store_found_cb(bind_this(LogDev::on_log_store_found, 2));
+    register_append_cb(bind_this(LogDev::on_io_completion, 5));
+    register_logfound_cb(bind_this(LogDev::on_logfound, 6));
```

Comment:
I think all these 3 callbacks can be removed. This was required because LogStoreFamily was separate class and thus register to LogDev. Now that is merged, it can simply be a function call and code will look lot simpler to follow.

Reply:
register_append_cb and register_append_cb are used by  test_log_dev to override . So kept the same.

Reply:
I think we should adjust this - It was inevitable earlier, now it is avoidable, so I feel doing a callback which only calls its local function is too confusing. We have the oppurtunity to simplify. So if you feel strongly of not doing it now, may be in next PR we can take care? I am not sure why test case really cares about internal callback. Am I missing something?

Reply:
Removed the callbacks.